tests/snowflake/operators/test_snowflake_async.py

```diff
@@ -53,46 +51,88 @@ def context():
-        self.dag = dag
-
-    @mock.patch(LONG_MOCK_PATH)
-    def test_snowflake_operator(self, mock_get_db_hook):
-        sql = """
-        CREATE TABLE IF NOT EXISTS test_airflow (
-            dummy VARCHAR(50)
-        );
-        """
-        operator = SnowflakeOperatorAsync(
-            task_id="basic_snowflake", sql=sql, dag=self.dag, do_xcom_push=False
-        )
-        # do_xcom_push=False because otherwise the XCom test will fail due to the mocking (it actually works)
-        # Dummy change
-        operator.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
-
-    @mock.patch(LONG_MOCK_PATH)
-    def test_snowflake_execute_operator_async(self, context):
-        """
-        Asserts that a task is deferred and an SnowflakeTrigger will be fired
-        when the SnowflakeOperatorAsync is executed.
-        """
-        sql = """
-        CREATE TABLE IF NOT EXISTS test_airflow (
-            dummy VARCHAR(50)
-        );
-        """
-
-        operator = SnowflakeOperatorAsync(
-            task_id="execute_run",
-            snowflake_conn_id=CONN_ID,
-            sql=sql,
+@mock.patch(LONG_MOCK_PATH)
+def test_snowflake_execute_operator_async(context):
+    """
+    Asserts that a task is deferred and an SnowflakeTrigger will be fired
+    when the SnowflakeOperatorAsync is executed.
+    """
+    sql = """
+    CREATE TABLE IF NOT EXISTS test_airflow (
+        dummy VARCHAR(50)
+    );
+    """
+
+    operator = SnowflakeOperatorAsync(
+        task_id="execute_run",
+        snowflake_conn_id=CONN_ID,
+        sql=sql,
+    )
+
+    with pytest.raises(TaskDeferred) as exc:
+        operator.execute(context)
+
+    assert isinstance(exc.value.trigger, SnowflakeTrigger), "Trigger is not a SnowflakeTrigger"
+
+
+def test_snowflake_async_execute_complete_failure():
+    """Tests that an AirflowException is raised in case of error event"""
+
+    sql = """
+            CREATE TABLE IF NOT EXISTS test_airflow (
+                dummy VARCHAR(50)
+            );
+            """
+
+    operator = SnowflakeOperatorAsync(
+        task_id="execute_complete",
+        snowflake_conn_id=CONN_ID,
+        sql=sql,
+    )
+    with pytest.raises(AirflowException):
+        operator.execute_complete(
+            context=None,
+            event={"status": "error", "message": "Test failure message", "type": "FAILED_WITH_ERROR"},
         )
 
-        with pytest.raises(TaskDeferred) as exc:
-            operator.execute(context)
 
-        assert isinstance(exc.value.trigger, SnowflakeTrigger), "Trigger is not a SnowflakeTrigger"
+@pytest.mark.parametrize(
+    "mock_event",
+    [
+        None,
+        ({"status": "success", "query_ids": ["uuid", "uuid"]}),
+    ],
+)
+@mock.patch(LONG_MOCK_PATH)
+def test_snowflake_async_execute_complete(mock_conn, mock_event):
+    sql = """
+            CREATE TABLE IF NOT EXISTS test_airflow (
+                dummy VARCHAR(50)
+            );
+            """
```

Comment:
same here

Reply:
Fixed

Reply:
This is still not fixed:

<img width="563" alt="image" src="https://user-images.githubusercontent.com/8811558/157534913-16bc07fd-0ed7-4037-8cbe-5b79b86d0b94.png">
vs
<img width="470" alt="image" src="https://user-images.githubusercontent.com/8811558/157534949-53d0bf3f-f19d-4a37-8237-ea6e919ce209.png">